plugins/circular-view/src/BaseChordDisplay/models/BaseChordDisplayModel.ts

```diff
@@ -0,0 +1,231 @@
+    },
+  }))
+  .actions(self => ({
+    renderStarted() {
+      self.filled = false
+      self.message = ''
+      self.html = ''
+      self.data = undefined
+      self.error = undefined
+      self.renderingComponent = undefined
+    },
+    renderSuccess({
+      message,
+      data,
+      html,
+      renderingComponent,
+    }: {
+      message: string
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      data: any
+      html: string
+      renderingComponent: AnyReactComponentType
+    }) {
+      if (message) {
+        self.filled = false
+        self.message = message
+        self.html = ''
+        self.data = undefined
+        self.error = undefined
+        self.renderingComponent = undefined
+      } else {
+        self.filled = true
+        self.message = ''
+        self.html = html
+        self.data = data
+        self.error = undefined
+        self.renderingComponent = renderingComponent
+      }
+    },
+    renderError(error: Error) {
+      console.error(error)
+      // the rendering failed for some reason
+      self.filled = false
+      self.message = ''
+      self.html = ''
+      self.data = undefined
+      self.error = error
+      self.renderingComponent = undefined
+    },
+
+    setRefNameMap(refNameMap: Record<string, string>) {
+      self.refNameMap = refNameMap
+    },
+    setError(error: Error) {
+      self.error = error
+    },
+  }))
+  .actions(self => {
+    const { pluginManager } = getSession(self)
+    const {
+      renderReactionData,
+      renderReactionEffect,
+    } = pluginManager.jbrequire(renderReactionFactory)
+    return {
+      afterAttach() {
+        makeAbortableReaction(
+          self,
+          renderReactionData,
+          renderReactionEffect,
+          {
+            name: `${self.type} ${self.id} rendering`,
+            // delay: self.renderDelay || 300,
+            fireImmediately: true,
+          },
+          self.renderStarted,
+          self.renderSuccess,
+          self.renderError,
+        )
+        makeAbortableReaction(
+          self,
+          () => ({
+            assemblyNames: getTrackAssemblyNames(self.parentTrack) as string[],
+            adapter: getConf(getParent(self, 2), 'adapter'),
+            assemblyManager: getSession(self).assemblyManager,
+          }),
+          // eslint-disable-next-line @typescript-eslint/no-explicit-any
+          async ({ assemblyNames, adapter, assemblyManager }: any, signal) => {
+            return assemblyManager.getRefNameMapForAdapter(
+              adapter,
+              assemblyNames[0],
+              { signal, sessionId: getRpcSessionId(self) },
+            )
+          },
+          {
+            name: `${self.type} ${self.id} getting refNames`,
+            fireImmediately: true,
+          },
+          () => {},
+          refNameMap => {
+            self.setRefNameMap(refNameMap)
+          },
+          error => {
+            self.setError(String(error))
```

Comment:
can we make this error instead of String(error). maybe add a console.error(error) here too